tiledb/sm/query/legacy/reader.cc

```diff
@@ -142,6 +142,66 @@ Reader::Reader(
   check_validity_buffer_sizes();
 }
 
+/* ********************************* */
+/*          STATIC FUNCTIONS         */
+/* ********************************* */
+
```

Comment:
Note: this just brings the functions that used to be in reader base into the legacy reader. The implementations between this and the dense reader diverged too much and only the code in the new ReaderBase::compute_tile_domains is still common.